pkg/telemetrymetadata/stmt_parse.go

```diff
@@ -0,0 +1,132 @@
+package telemetrymetadata
+
+import (
+	"strings"
+
+	"github.com/AfterShip/clickhouse-sql-parser/parser"
+	"github.com/SigNoz/signoz/pkg/types/telemetrytypes"
+)
+
+// TelemetryFieldVisitor is an AST visitor for extracting telemetry fields
+type TelemetryFieldVisitor struct {
+	parser.DefaultASTVisitor
+	Fields []*telemetrytypes.TelemetryFieldKey
+}
+
+func NewTelemetryFieldVisitor() *TelemetryFieldVisitor {
+	return &TelemetryFieldVisitor{
+		Fields: make([]*telemetrytypes.TelemetryFieldKey, 0),
+	}
+}
+
+// VisitColumnDef is called when visiting a column definition
+func (v *TelemetryFieldVisitor) VisitColumnDef(expr *parser.ColumnDef) error {
+	// Check if this is a materialized column with DEFAULT expression
```

Comment:
I don't get this comment

Reply:
Every materialized column in our case would have the default expression. We look at the default expression to find the attribute key being materialized. Please take a look at the test and table schema.

Reply:
Got the point now.